plugins/pulp_docker/plugins/importers/upload.py

```diff
@@ -169,6 +179,47 @@ def get_models(self, metadata, mask_id=''):
         return images
 
 
+class ImportManifestList(PluginStep):
+    """
+    Add an uploaded ManifestList to a repository.
+    """
+
+    def process_main(self):
+        """
+        Validate the uploaded manifest list json, then import content unit into repository.
+        """
+        with open(self.parent.file_path, 'r') as uploaded_file:
+            manifest_list = uploaded_file.read()
+        models.ManifestList.validate_json(manifest_list)
+        digest = models.Manifest.calculate_digest(manifest_list)
+        manifest_list_instance = models.ManifestList.from_json(manifest_list, digest)
+        transfer_repo = self.get_repo()
+
+        #  Ensure that all referenced manifests are already in repository.
+        manifest_digests = set()
+        for manifest in manifest_list_instance.manifests:
```

Comment:
since manifests is a list i guess you can make directly set from it  manifest_digests = set(manifest_list_instance.manifests)